Source/WebCore/platform/mediarecorder/cocoa/AudioSampleBufferCompressor.mm

```diff
@@ -534,12 +534,9 @@
 
 void AudioSampleBufferCompressor::addSampleBuffer(CMSampleBufferRef buffer)
 {
-    // Heap allocations are forbidden on the audio thread for performance reasons so we need to
-    // explicitly allow the following allocation(s).
-    DisableMallocRestrictionsForCurrentThreadScope disableMallocRestrictions;
```

Comment:
We should probably keep these if we want to go back to WebProcess.

Reply:
My plan was to have the MediaRecorderPrivateWriter be in the main thread instead. so this would need to be in the thread proxy between the `RealtimeMediaSource::AudioSampleObserver` and `RealtimeMediaSource::VideoFrameObserver` and its MediaRecorderPrivate ; in which case it's there that this is needed.

Reply:
I've re-instated for now. This method is currently only ever called on the main thread